### PR TITLE
feat: sync inspect mpt tool from bsc

### PR DESCRIPTION
**Pull request by will-2012**
### Description

Cherry-pick inspect mpt tool from bsc.

https://github.com/bnb-chain/bsc/pull/2082
https://github.com/bnb-chain/bsc/pull/2142

### Rationale

N/A.

### Example

N/A.

### Changes

Notable changes:
* CMD: add inspect mpt tool.

### CODE REVIEW

File: cmd/geth/dbcmd.go [hunk 1]
```diff
@@ -255,6 +268,82 @@ func confirmAndRemoveDB(database string, kind string) {
...
+			jobnum, err = strconv.ParseUint(ctx.Args().Get(1), 10, 64)
+			if err != nil {
+				return fmt.Errorf("failed to Parse jobnum, Args[1]: %v, err: %v", ctx.Args().Get(1), err)
+			}
+		}
+
+		if blockNumber != math.MaxUint64 {
+			headerBlockHash = rawdb.ReadCanonicalHash(db, blockNumber)
+			if headerBlockHash == (common.Hash{}) {
+				return fmt.Errorf("ReadHeadBlockHash empry hash")
```
[the remark this answers is not in the document]
Comment (will-2012) [reply cmd/geth/dbcmd.go]: Fixed.

**Review by nolanxyg — APPROVED** [review]

**Review by bendanzhentan — APPROVED** [review]

File: trie/inspect_trie.go [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+	totalNum       uint64
+	wg             sync.WaitGroup
+	statLock       sync.RWMutex
+	result         map[string]*TrieTreeStat
+	sem            *semaphore.Weighted
+	eoaAccountNums uint64
+}
+
+type TrieTreeStat struct {
+	is_account_trie    bool
```
Comment (bendanzhentan) [inline trie/inspect_trie.go:52]: should we use camel naming style?

Comment (will-2012) [reply trie/inspect_trie.go]: Fixed~

**Review by RenRick — APPROVED** [review]
LGTM

File: trie/inspect_trie.go [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/olekukonko/tablewriter"
+
```
Comment (sysvm) [inline trie/inspect_trie.go:20]: Remove the blank line?

**Review by owen-reorg — APPROVED** [review]

**Pull request merged**